### PR TITLE
SOC-1161: Followed Forum and Wall threads should be bold on RecentChanges

### PR DESCRIPTION
**Pull request by mszabo-wikia**
@garthwebb here you go :)

### CODE REVIEW

File: extensions/wikia/ArticleComments/classes/ArticleComment.class.php [hunk 1]
```diff
@@ -1113,20 +1113,20 @@ static public function addArticlePageToWatchlist( $comment ) {
 	 *
 	 * @return Bool true -- because it's a hook
 	 */
-	static public function watchlistNotify( RecentChange &$oRC ) {
-		global $wgEnableGroupedArticleCommentsRC;
-		wfProfileIn( __METHOD__ );
+	static public function watchlistNotify(RecentChange &$oRC) {
```
Comment (kvas-damian) [inline extensions/wikia/ArticleComments/classes/ArticleComment.class.php]: Please follow MW coding conventions about spaces: https://www.mediawiki.org/wiki/Manual:Coding_conventions/PHP#Spaces

File: extensions/wikia/Wall/WallHooksHelper.class.php [hunk 2]
```diff
@@ -902,20 +927,26 @@ static public function onChangesListInsertArticleLink( $list, &$articleLink, &$s
 				$title = $wm->getMetaTitle();
 				$titleText = $wm->getArticleTitle()->getPrefixedText();
 				$pageText = $wm->getMainPageText();
-				$class = '';
 
-				$articleLink = ' <a href="' . $link . '" class="' . $class . '" >' . $title . '</a> ' . wfMessage( static::getMessagePrefix( $rc->getAttribute( 'rc_namespace' ) ) . '-new-message', array( $titleText, $pageText ) )->parse();
+				$attribs = [
+					'class' => '',
```
Comment (kvas-damian) [inline extensions/wikia/Wall/WallHooksHelper.class.php]: do we need this attribute?

File: extensions/wikia/Wall/WallHooksHelper.class.php [hunk 3]
```diff
@@ -1310,6 +1341,15 @@ static public function onWikiaRecentChangesBlockHandlerChangeHeaderBlockGroup( $
 				->rawParams( $threadLink )
 				->params( $titleData['wallTitleTxt'], $titleData['wallPageName'] )
 				->parse();
+
+			// VOLDEV-3: Bolden pages watched by this user
+			$user = $oChangeList->getUser();
+			$specialPageTitle = $oChangeList->getTitle();
+			if ( self::boldenFollowedLink( $user, $specialPageTitle, WallMessage::newFromTitle( $oTitle ) ) ) {
+				$headerTitle = '<strong class="mw-watched">' . $headerTitle . '</strong>';
```
Comment (kvas-damian) [inline extensions/wikia/Wall/WallHooksHelper.class.php]: This line is duplicated in this file. What do you think about extracting it to separate method?

File: extensions/wikia/Wall/WallHooksHelper.class.php [hunk 4]
```diff
@@ -2213,6 +2258,24 @@ public static function onArticleCommentGetSquidURLs( $title, &$urls ) {
...
 	/**
+	 * Determine whether a link pointing to a followed thread should be bold
+	 * A link should be bold if it's followed by the user, we are on Special:Watchlist
+	 * and the page was changed since the user's last visit
+	 * @param User $user
+	 * @param Title $title Title of the special page (RecentChanges or Watchlist)
+	 * @param WallMessage $wm
+	 * @return bool Whether to bolden the link
+	 */
+	private static function boldenFollowedLink( User $user, Title $title, WallMessage $wm ) {
```
Comment (kvas-damian) [inline extensions/wikia/Wall/WallHooksHelper.class.php]: What do you think about making this method name more descriptive?

```
shouldBoldenFollowedLink
```

?

**Comment by mszabo-wikia** [comment]
@kvas-damian Thanks for your review! Done.

**Comment by kvas-damian** [comment]
LGTM, It's time for QA now ;)

**Pull request merged**